### PR TITLE
Remove ansi codes from terminal output

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>

##### SUMMARY
Fixes : https://github.com/ansible-collections/vyos.vyos/issues/65    and    https://github.com/ansible-collections/vyos.vyos/issues/66

This can be removed once https://github.com/ansible/ansible/pull/70683 is available.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
plugins/terminal/vyos.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by mjrider** [comment]
Sorry to report that this didn't help for me
ansible 2.9.10 for if that matters, starts to feel like i'm doing something wrong with testing all these potential fixes.

i'll try to verify that i'm not doing something wrong tomorrow

**Comment by Qalthos** [comment]
I've tried to match your environment as closely as possible here.

Using `network_os=vyos` it uses the terminal plugin shipped with ansible 2.9:
```
% ansible-playbook vyos_facts.yaml -e "ansible_network_os=vyos" -vv
ansible-playbook 2.9.10
  config file = /home/nate/workspace/ansible/playbooks/test/ansible.cfg
  configured module search path = [u'/home/nate/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nate/.local/virtualenvs-Arch/ansible2/lib/python2.7/site-packages/ansible
  executable location = /home/nate/.local/virtualenvs-Arch/ansible2/bin/ansible-playbook
  python version = 2.7.18 (default, Apr 23 2020, 22:32:06) [GCC 9.3.0]
Using /home/nate/workspace/ansible/playbooks/test/ansible.cfg as config file

PLAYBOOK: vyos_facts.yaml *********************************************************************************************
1 plays in vyos_facts.yaml

PLAY [vyos] ***********************************************************************************************************
META: ran handlers

TASK [vyos.vyos.vyos_facts] *******************************************************************************************
task path: /home/nate/workspace/ansible/playbooks/test/vyos_facts.yaml:7
[WARNING]: Persistent connection logging is enabled for vyos.linkybook. This will log ALL interactions to
/tmp/ansible.log and WILL NOT redact sensitive configuration like passwords. USE WITH CAUTION!
ok: [vyos.linkybook] => changed=false 
  ansible_facts:
    ansible_net_api: cliconf
    ansible_net_gather_network_resources: []
    ansible_net_gather_subset:
    - default
    ansible_net_hostname: "vyos\e[m"
    ansible_net_python_version: 2.7.18
    ansible_net_serialnum: null
    ansible_net_system: vyos
    ansible_net_version: "VyOS 1.3-rolling-202007130117\e[m"
    ansible_network_resources: {}
META: ran handlers
META: ran handlers

PLAY RECAP ************************************************************************************************************
vyos.linkybook             : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Using `network_os=vyos.vyos.vyos` (oof) it picks up the terminal plugin from the collection:
```% ansible-playbook vyos_facts.yaml -e "ansible_network_os=vyos.vyos.vyos" -vv
ansible-playbook 2.9.10
  config file = /home/nate/workspace/ansible/playbooks/test/ansible.cfg
  configured module search path = [u'/home/nate/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nate/.local/virtualenvs-Arch/ansible2/lib/python2.7/site-packages/ansible
  executable location = /home/nate/.local/virtualenvs-Arch/ansible2/bin/ansible-playbook
  python version = 2.7.18 (default, Apr 23 2020, 22:32:06) [GCC 9.3.0]
Using /home/nate/workspace/ansible/playbooks/test/ansible.cfg as config file

PLAYBOOK: vyos_facts.yaml *********************************************************************************************
1 plays in vyos_facts.yaml

PLAY [vyos] ***********************************************************************************************************
META: ran handlers

TASK [vyos.vyos.vyos_facts] *******************************************************************************************
task path: /home/nate/workspace/ansible/playbooks/test/vyos_facts.yaml:7
[WARNING]: Persistent connection logging is enabled for vyos.linkybook. This will log ALL interactions to
/tmp/ansible.log and WILL NOT redact sensitive configuration like passwords. USE WITH CAUTION!
ok: [vyos.linkybook] => changed=false 
  ansible_facts:
    ansible_net_api: cliconf
    ansible_net_gather_network_resources: []
    ansible_net_gather_subset:
    - default
    ansible_net_hostname: vyos
    ansible_net_python_version: 2.7.18
    ansible_net_serialnum: null
    ansible_net_system: vyos
    ansible_net_version: VyOS 1.3-rolling-202007130117
    ansible_network_resources: {}
META: ran handlers
META: ran handlers

PLAY RECAP ************************************************************************************************************
vyos.linkybook             : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

My guess is you haven't updated `network_os` to use the [FQCN](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-collections-in-a-playbook) (which is pretty understandable, this is still pretty new and confusing). Hopefully this will get easier in the nearish future when you either start using 2.10 (and don't have two copies of the content anymore) or just get used to putting FQCNs everywhere, whichever comes first

File: changelogs/fragments/65-remove-unwanted-terminal-chars.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - Added characters to be removed from terminal output.
```
Comment (Qalthos) [inline changelogs/fragments/65-remove-unwanted-terminal-chars.yaml:3]:
```suggestion
  - terminal plugin - Added additional escape sequence to be removed from terminal output.
```

File: plugins/terminal/vyos.py [hunk 2]
```diff
@@ -40,6 +40,12 @@ class TerminalModule(TerminalBase):
         re.compile(br"\n\s+Set failed"),
     ]
 
+    ansi_re = [
+        re.compile(br'\x1b\[\?1h\x1b='),  # CSI ? 1 h ESC =
+        re.compile(br'\x08.'),            # [Backspace] .
```
Comment (Qalthos) [inline plugins/terminal/vyos.py:45]:
```suggestion
        re.compile(br"\x1b\[\?1h\x1b="),  # CSI ? 1 h ESC =
        re.compile(br"\x08.'),            # [Backspace] .
```

**Comment by mjrider** [comment]
@Qalthos Thx, needed a bit of coffee to figure out that it was `ansible_network_os` but yes, that works.

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Pull request merged**